### PR TITLE
Enable image replacement 

### PR DESCRIPTION
**Pull request by vincent-pli**
I think we need this ability, as I talked in this issue: #54 
I check the implements in `knative-serving operator` and implement in tekton.

### CODE REVIEW

**Comment by nikhil-thomas** [comment]
https://github.com/vincent-pli we setup CI with e2e tests. Could you rebase your branch.

**Comment by vincent-pli** [comment]
@nikhil-thomas @sthaha 
Is the e2e failure is a real broken?
Could you take a look at it?

**Comment by sthaha** [comment]
@vincent-pli thank you for your patience. Since #66 has now merged, could you please rebase this patch?

File: pkg/controller/config/add_imagereplacement.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
+/*
+Copyright 2019 The Knative Authors
```
Comment (sthaha) [inline pkg/controller/config/add_imagereplacement.go]: :) Tekton

File: pkg/controller/config/common/extensions.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2019 The Knative Authors
```
Comment (sthaha) [inline pkg/controller/config/common/extensions.go]: same - Tekton

File: pkg/controller/config/common/extensions.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+type Activities []func(client.Client, *runtime.Scheme, *tektonv1alpha1.Config) (*Extension, error)
+type Extender func(*tektonv1alpha1.Config) error
+type Extensions []Extension
+type Extension struct {
+	Transformers []mf.Transformer
+	PreInstalls  []Extender
+	PostInstalls []Extender
+}
+
+func (activities Activities) Extend(c client.Client, scheme *runtime.Scheme, install *tektonv1alpha1.Config) (result Extensions, err error) {
```
Comment (nikhil-thomas) [inline pkg/controller/config/common/extensions.go]: `install` change this name as our CRD is not `install.tekton.dev` anymore

File: pkg/controller/config/extensions/imagereplacement/helpers_test.go [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+	expecteds = expecteds[1:]
+	// Assert container[*].args
+	for i, argsIndex := range tt.argsIndex {
+		assertEqual(t, deployment.Spec.Template.Spec.Containers[0].Args[argsIndex], expecteds[i])
+	}
+
+}
+
+func assertEqual(t *testing.T, actual, expected string) {
+	if actual == expected {
```
Comment (nikhil-thomas) [inline pkg/controller/config/extensions/imagereplacement/helpers_test.go]: wouldn't it be better to add
`t.Helper()` as the first line in the function `assertEqual`

File: pkg/controller/config/common/extensions.go (hunk 3 above)
Comment (vincent-pli) [reply pkg/controller/config/common/extensions.go]: ok, change to `config`?

File: pkg/controller/config/extensions/imagereplacement/helpers_test.go (hunk 4 above)
Comment (vincent-pli) [reply pkg/controller/config/extensions/imagereplacement/helpers_test.go]: ok

**Comment by sthaha** [comment]
@vincent-pli closing this PR. lets continue this upstream of needed.

**Pull request closed without merging**